test/ILLink.RoslynAnalyzer.Tests/AvoidAssemblyLocationInSingleFileTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = ILLink.RoslynAnalyzer.Tests.CSharpAnalyzerVerifier<
+	ILLink.RoslynAnalyzer.AvoidAssemblyLocationInSingleFile>;
+
+namespace ILLink.RoslynAnalyzer.Tests
+{
+	public class AvoidAssemblyLocationInSingleFileTests
+	{
+		[Fact]
+		public Task GetExecutingAssemblyLocation ()
+		{
+			const string src = @"
+using System.Reflection;
+class C
+{
+    public string M() => Assembly.GetExecutingAssembly().Location;
+}";
+
+			return VerifyCS.VerifyAnalyzerAsync (src,
+				new (string, string)[] { ($"build_property.{MSBuildPropertyOptionNames.PublishSingleFile}", "true") },
```

Comment:
I'd add a helper method in this class to pass `PublishSingleFile` as true (since almost every test passes it)